Dockerfile

```diff
@@ -25,16 +25,26 @@ RUN set -o pipefail && \
   wget -qO - "${SBT_URL}" | tar xz -C "${INSTALL_DIR}" && \
   echo -ne "- with sbt ${SBT_VERSION}\n" >> /root/.built
 
+RUN mkdir -p "${PROJECT_LOC}"
+WORKDIR "${PROJECT_LOC}"
+
+RUN sbt update
+
+COPY "${PROJECT_NAME}"/package* .
+RUN npm install
+
+# Don't copy the project source until here, so everything above gets cached.
 COPY "${PROJECT_NAME}" "${PROJECT_LOC}"
```

Comment:
Tbh I don't understand it. Wouldn't copying to the PROJECT_LOC overwrite everything that that folder had until that call?

Reply:
Yes, but that's not a problem.  We are using the npm cache when downloading the files, and entrypoint re-runs npm install. And we just need to run sbt so that it downloads all it's own dependancies.